qiskit/pulse/timeslots.py

```diff
@@ -193,68 +270,163 @@ def duration(self) -> int:
         """Return maximum duration of timeslots over all channels."""
         return self.stop_time
 
-    def ch_start_time(self, *channels: List[Channel]) -> int:
+    def _merge_timeslot_collection(self, other: 'TimeslotCollection'):
+        """Mutably merge timeslot collections into this TimeslotCollection.
+
+        Args:
+            other: TimeSlotCollection to merge
+        """
+        common_channels = set(self.channels) & set(other.channels)
+
+        for channel in other.channels:
+            ch_timeslots = self._table[channel]
+            other_ch_timeslots = other._table[channel]
+            # if channel is in self there might be an overlap
+            timeslot_idx = 0
+            if channel in common_channels:
+                for other_ch_timeslot in other_ch_timeslots:
+                    insert_idx = self._merge_timeslot(other_ch_timeslot)
+
+                    timeslot_idx += 1
+                    # timeslot was inserted at end of list. Remaining timeslots can be appended.
+                    if insert_idx == len(self._table[channel]) - 1:
+                        break
+
+                ch_to_append = other_ch_timeslots[timeslot_idx:]
+
+            # otherwise directly insert
+            else:
+                ch_to_append = other_ch_timeslots
+
+            if ch_to_append:
+                ch_timeslots += ch_to_append
+
+    def _merge_timeslot(self, timeslot: Timeslot) -> int:
+        """Mutably merge timeslots into this TimeslotCollection.
+
+        Note timeslots are sorted internally on their respective channel
+
+        Args:
+            timeslot: Timeslot to merge
+
+        Returns:
+            int: Return the index in which timeslot was inserted
+
+        Raises:
+            PulseError: If timeslots overlap
+        """
+        interval = timeslot.interval
+        ch_timeslots = self._table[timeslot.channel]
+
+        insert_idx = len(ch_timeslots)
+
+        # bubble sort for insertion location.
+        # Worst case O(n_channels), O(1) for append
```

Comment:
This is insignificant, but this sort is more like `insert sort` than`bubble sort`. And this is also no wrong, but I think simple binary search (key=interval.start), instead of interval tree, would be enough in this case since intervals are sorted and have non-overlap.